src/bio/bio_xstream.c

```diff
@@ -1771,35 +1775,34 @@ scan_bio_bdevs(struct bio_xs_context *ctxt, uint64_t now)
 	struct bio_blobstore	*bbs;
 	struct bio_bdev		*d_bdev, *tmp;
 	struct spdk_bdev	*bdev;
+	const char		*bdev_name;
 	static uint64_t		 scan_period = NVME_MONITOR_PERIOD;
 	int			 rc;
 
 	if (nvme_glb.bd_scan_age + scan_period >= now)
 		return;
 
 	/* Iterate SPDK bdevs to detect hot plugged device */
-	for (bdev = spdk_bdev_first(); bdev != NULL;
-	     bdev = spdk_bdev_next(bdev)) {
+	for (bdev = spdk_bdev_first(); bdev != NULL; bdev = spdk_bdev_next(bdev)) {
 		if (nvme_glb.bd_bdev_class != get_bdev_type(bdev))
 			continue;
 
-		d_bdev = lookup_dev_by_name(spdk_bdev_get_name(bdev));
+		bdev_name = spdk_bdev_get_name(bdev);
+
+		d_bdev = lookup_dev_by_name(bdev_name);
 		if (d_bdev != NULL)
 			continue;
 
-		D_INFO("Detected hot plugged device %s\n",
-		       spdk_bdev_get_name(bdev));
+		D_INFO("Detected hot plugged device %s\n", bdev_name);
 		/* Print a console message */
-		D_PRINT("Detected hot plugged device %s\n",
-			spdk_bdev_get_name(bdev));
+		D_PRINT("Detected hot plugged device %s\n", bdev_name);
 
 		scan_period = 0;
 
-		/* don't support hot plug for sys device yet */
-		rc = create_bio_bdev(ctxt, spdk_bdev_get_name(bdev), &d_bdev);
+		/* don't support hot plug for sys device yet, assign default roles */
+		rc = create_bio_bdev(ctxt, bdev_name, NVME_ROLE_DATA, &d_bdev);
```

Comment:
This should pass in bdev_name2roles(bdev_name)?

Reply:
This is hot plugged device, bdev_name2roles() won't for such device name.

Reply:
oops, indeed.